pkg/version/version.go

```diff
@@ -77,3 +80,77 @@ func (v *Build) String() string {
 func (v *Build) ShortVersion() string {
 	return fmt.Sprintf("%s-%s-%s", v.Version, v.BuildNumber, v.GitCommit)
 }
+
+// Equal determines if v is equal to b based on Version and BuildNumber
+func (v *Build) Equal(b *Build) bool {
+	if v.Version != b.Version || v.BuildNumber != b.BuildNumber {
+		return false
+	}
+	return true
+}
+
+// IsOlder determines if v is older than b based on Version and BuildNumber
+func (v *Build) IsOlder(b *Build) (bool, error) {
+	if v.Equal(b) {
+		return false, nil
+	}
+	verBefore, err := compareVersionString(v.Version, b.Version)
+	if err != nil {
+		return false, err
+	}
+
+	// version short circuits
+	if verBefore {
+		return true, nil
+	}
+
+	if v.BuildNumber == "<nil>" || b.BuildNumber == "<nil>" {
+		return false, errors.New("invalid BuildNumber")
+	}
+	buildBefore := v.BuildNumber < b.BuildNumber
+	return buildBefore, nil
+}
+
+// IsNewer determines if v is newer than b based on Version and BuildNumber
+func (v *Build) IsNewer(b *Build) (bool, error) {
+	if v.Equal(b) {
+		return false, nil
+	}
+	older, err := v.IsOlder(b)
+	if err != nil {
+		return false, err
+	}
+	return !older, nil
+}
+
+// compareVersionString returns if Version string (e.g. v1.2.3) of v is before that of b
+// ignores pre-release identifiers (e.g. v1.2.3-alpha)
+func compareVersionString(v, b string) (bool, error) {
+	if !strings.HasPrefix(v, "v") || !strings.HasPrefix(b, "v") {
```

Comment:
this method assume the version has same format, but we cannot make such assumption.
So basically, we just use buildnumber to detect which one is earlier.